File: 27-06-2025/p2014.cpp
class Solution {
public:
bool find(string s,string p,int k){
    int i=0;
    int j=0;
    while(i<s.size() && k){
        if(s[i]==p[j]){
            i++;
            j++;
        }
        else
        i++;
        if(j==p.size()){
            j=0;
        k--;
        }

    }
    return k<=0;
}
    string longestSubsequenceRepeatedK(string s, int k) {
        queue<string>q;
        q.push("");
        string ans;
        while(!q.empty()){
            auto str=q.front();
            q.pop();
            string p=str;
            for(char ch='a';ch<='z';ch++){
                if(find(s,str+ch,k)){
                    p=str+ch;
                    if(p.size()>ans.size() || p>ans)
                    ans=p;
                    q.push(p);
                }
                
            }
        }
        return ans;

    }
};

File: JUNE-2025/27-06-2025/p2014.php
class Solution {

    /**
     * @param String $s
     * @param Integer $k
     * @return String
     */
    function longestSubsequenceRepeatedK($s, $k) {
         $ans = "";
        $queue = new SplQueue();
        $queue->enqueue("");

        while (!$queue->isEmpty()) {
            $curr = $queue->dequeue();
            for ($ch = ord('a'); $ch <= ord('z'); $ch++) {
                $next = $curr . chr($ch);
                if ($this->solve($next, $s, $k)) {
                    $ans = $next;
                    $queue->enqueue($next);
                }
            }
        }

        return $ans;
    }

     public function solve($sub, $s, $k) {
        $i = 0;
        $count = 0;
        $subLen = strlen($sub);

        for ($j = 0; $j < strlen($s); $j++) {
            if ($s[$j] === $sub[$i]) {
                $i++;
                if ($i === $subLen) {
                    $i = 0;
                    $count++;
                    if ($count === $k) {
                        return true;
                    }
                }
            }
        }
        return false;
    }
}